kotlin_spark/src/main/java/com/sleazyweasel/eboshi/Auth.kt

```diff
@@ -0,0 +1,68 @@
+package com.sleazyweasel.eboshi
+
+import org.jose4j.jwk.RsaJsonWebKey
+import org.jose4j.jwk.RsaJwkGenerator
+import org.jose4j.jws.AlgorithmIdentifiers
+import org.jose4j.jws.JsonWebSignature
+import org.jose4j.jwt.JwtClaims
+import org.jose4j.jwt.consumer.JwtConsumerBuilder
+import org.mindrot.jbcrypt.BCrypt
+
+private val TOKEN_ISSUER = "Kotlin Eboshi API"
+private val TOKEN_AUDIENCE = "eboshi user"
+private val EMAIL_CLAIM_KEY = "email"
+
+open class Auth {
+    private val rsaWebKey: RsaJsonWebKey;
+
+    constructor() {
+        rsaWebKey = RsaJwkGenerator.generateJwk(2048)!!
```

Comment:
Or wait, this is in the constructor... does that mean that there is a singleton Auth? It looks like its being created by the injector, when the AccountRoutes object is created. So... yes?